core/database/arango/optimized_client.py

```
@@ -0,0 +1,196 @@
+    def __init__(self, status_code: int, message: str, details: dict[str, Any] | None = None) -> None:
+        super().__init__(f"ArangoDB HTTP {status_code}: {message}")
+        self.status_code = status_code
+        self.details = details or {}
+
+
+@dataclass
+class ArangoHttp2Config:
+    """Configuration for the HTTP/2 client."""
+
+    database: str = "_system"
+    socket_path: str = "/tmp/arangodb.sock"
```

Ruff (0.13.1)

Probable insecure usage of temporary file or directory: "/tmp/arangodb.sock"

(S108)